src/actions.cpp

```diff
@@ -172,27 +172,19 @@ namespace Action {
         return UniquePtr(clone_());
     }
 
-    TaskFactory* TaskFactory::instance_ = 0;
-
     TaskFactory& TaskFactory::instance()
     {
-        if (0 == instance_) {
-            instance_ = new TaskFactory;
-        }
-        return *instance_;
-    } // TaskFactory::instance
+        static TaskFactory ins;
+        return ins;
+    }
 
     void TaskFactory::cleanup()
```

Comment:
Shouldn't this go into the destructor instead?

Reply:
Right. Let's do that